src/Searchlight/DataSource.cs

```diff
@@ -269,6 +269,11 @@ public SyntaxTree Parse(FetchRequest request)
                 {
                     throw new InvalidPageSize { PageSize = request.pageSize == null ? "not specified" : request.pageSize.ToString() };
                 }
+                
+                if (SearchlightEngine.MaximumPageSize > 1 & query.PageSize > SearchlightEngine.MaximumPageSize)
+                {
+                    throw new InvalidPageSize { PageSize = request.pageSize == null ? "not specified" : "larger than the allowed MaximumPageSize" };
```

Comment:
Please include, in the text of this message, what the maximum page size is.  As a rule, every error must explain not only what went wrong but what must change in order to make it right.